### DISCUSSION TITLE
Custom Theme App.css not loading on build or dev.

### DISCUSSION THREAD

**Discussion by CoconutDev-ops** [filamentphp/filament#5956] [Help]
Hello,

I'm having a bit trouble creating my own theme, I'm pretty new to Filament so learning the ropes a little. I've followed the documentation guide on how to create my own theme, however when doing so the entirety of my application becomes un-styled. If I roll back the edits I've made to the theme it all returns to normal.

When inspecting the application from the front end, when everything is left like an original, clean install the path "filament/assets/app.css" is loading into the "source" tab, when I implement the custom theme the filament/assets/app.css disappears, and when it does it takes all the styling away.

Here is a screenshot of the sources tab when I've got a 'custom theme' in place:
![image](https://user-images.githubusercontent.com/69913629/225942024-d8aae614-f842-472a-ada1-5fc9fd51f295.png)
![image](https://user-images.githubusercontent.com/69913629/225942447-118ab48b-c8e0-4ed0-ac25-062274509d5e.png)

And here it is when I've rolled back everything and using the default code I got:
![image](https://user-images.githubusercontent.com/69913629/225943115-537f2e69-05e9-4699-831b-505e73ba158d.png)

Anyone have any idea what is going wrong? What have I missed? Any advise or guidance is appreciated.

I'm on Laravel (latest) and using VITE to build the css.

**Comment by danharrin** [comment]
Does the compiled CSS file exist in `/public`?

**Reply by CoconutDev-ops** [reply]
Hello @danharrin, it doesn't exist in the /public folder. It get generated if I do /npm run build. I've also tried to manually place the css inside the /public folder both inside a folder (/css) and as single file, there was no changes to the front end.

**Reply by danharrin** [reply]
Unless it is compiled into `/public`, Filament isn't going to be able to serve it to any users, right? The default Vite destination should be inside `/public` somewhere.

**Reply by CoconutDev-ops** [reply]
Hi @danharrin, thanks for the reply. I'm a clear on the need to have the css in the /public directory, I'm just a little bewildered as to why it's not pulling through. 

Can you recommend any steps to try a fix it? I've tried various from the little I've found online.

**Reply by danharrin** [reply]
I would only be repeating the steps in the docs. If other people had problems following then I would probably revisit those steps and check them, but it seems fine.

**Reply by damann8479** [reply]
I have the same problem. were you able to find a solution?

**Reply by tapanbasuli** [reply]
If you are using ngnix remove **css|js|** from

`location ~* ^.+\.(css|js|jpg|jpeg|gif|png|ico|gz|svg|svgz|ttf|otf|woff|woff2|eot|mp4|ogg|ogv|webm|webp|zip|swf|map)$`

**Comment by christoferd** [comment]
For others troubleshooting the same issue, what fixed it for me was that I forgot to add this at the top of tailwind config...

```const colors = require('tailwindcss/colors')```

Also, try to run: ```npm run build```
instead of npm run dev to see if it works.

**Comment by damann8479** [comment]
Hello, i have the same problem. i foolowes the instructions. 
https://filamentphp.com/docs/2.x/admin/appearance#building-themes

**Comment by BryarAhmed** [comment]
I have this problem too.

This problem is caused by registerTheme in the ServiceProvider in here:

Filament::serving(function () {
    
// Caused by this, it's not working   Filament::registerViteTheme('resources/css/filament.css');


// I have tried this too and it's the same, it's not working 
Filament::registerTheme(
            app(Vite::class)('resources/css/app.css'),
        );
 
});


any idea please, is there any way to do this in a different way?

**Reply by aliaswpeu** [reply]
I think You should try register filament.css not app.css

**Comment by iamvincentong** [comment]
I faced the same problem, but I found a solution. Here are the steps I took:

1. Install Tailwind and its plugins by running this command:
`npm install tailwindcss @tailwindcss/forms @tailwindcss/typography postcss autoprefixer tippy.js --save-dev`

2. Create two new files, tailwind.config.js and postcss.config.js, in the main folder of your project. You can do this by running the following command:
`npx tailwindcss init && touch postcss.config.js`

3. In the tailwind.config.js file, include the plugins you installed and add any custom colors that you want to use for the form builder.

4. Add Tailwind to the postcss.config.js file:
`export default {
    plugins: {
        tailwindcss: {},
        autoprefixer: {},
    },
}
`

5. Finish the remaining steps, just follow the instructions given in the documentation:
https://filamentphp.com/docs/2.x/admin/appearance#building-themes

**Comment by CoconutDev-ops** [comment]
Sorry everyone for not responding sooner. The fix I used was not including the filament.css inside the resource/css folder. 

Following the guide works, just make sure you take every step slowly so nothing gets missed or overlooked, that was my issue.

**Comment by vorosarmand** [comment]
I finally found the solution for me. I needed to place the following inside my postcss.config.js:

module.exports = { plugins: { tailwindcss: {}, autoprefixer: {} } };

And it worked..

**Comment by tomcoonen** [comment]
@danharrin Seems like on the v3 documentation adding the `postcss.config.js` step is missing? Or am I missing something?

**Reply by danharrin** [reply]
The `make:filament-theme` command creates that file if it doesn't exist

**Reply by tomcoonen** [reply]
Ok thanks. For some reason I was missing the file then, If I run into it again I'll dive into it.
